### ISSUE TITLE
Docker Desktop Mac 4.15.0 has some warning messages running the build-tools container

### ISSUE THREAD

**Issue by ericvn** [istio/istio#42240]
### Bug Description

Moving to Docker Desktop on the Mac, 4.15.0 causes a few warning messages on the `make` commands, although they still seem to work:
```
$ make shell
cp: cannot stat '/config/.docker/run/docker.sock': No such file or directory
cp: cannot stat '/config/.docker/run/docker-cli-api.sock': No such file or directory
docker build-tools:/work$ docker ps
CONTAINER ID   IMAGE                                                                              COMMAND                  CREATED         STATUS        PORTS     NAMES
3212240c09cb   gcr.io/istio-testing/build-tools:master-4bebfc2f3383c6e535409ad31559c18fe36a4182   "/usr/local/bin/dock…"   2 seconds ago   Up 1 second             nostalgic_golick
```

```
ls -la /config/.docker/run
ls: cannot access '/config/.docker/run/docker-cli-api.sock': No such file or directory
ls: cannot access '/config/.docker/run/docker.sock': No such file or directory
total 0
drwxr-xr-x  4 user root 128 Dec  1 17:08 .
drwxr-xr-x 16 user root 512 Dec  1 16:20 ..
s?????????  ? ?    ?      ?            ? docker-cli-api.sock
s?????????  ? ?    ?      ?            ? docker.sock
```

Likely: https://github.com/istio/tools/blob/master/docker/build-tools/docker-entrypoint.sh#L24. I'm not sure if we can simply ignore the `run` sub-directory, but the `config-copy/.docker` and `home/.docker` don't have the files today so suspect it would. Maybe set GLOBIGNORE to docker*.sock

### Version

```prose
Using the main branch.
```

**Comment by ericvn** [comment]
Comment added in Docker GitHub issue: https://github.com/docker/roadmap/issues/7#issuecomment-1341158223

**Comment by fredericdalleau** [comment]
@ericvn socket support should be available in MacOS 13. What version are you running?

**Comment by ericvn** [comment]
Changing the `cp` to an `rsync` which ignores the dock*.sock files should and does remove the warnings.

**Comment by ericvn** [comment]
They may have fixed the message in Docker Desktop for Mac when testing on a newer version (also fixes an executable bit problem) so closing this for now.

**Comment by ericvn** [comment]
Trying again on the 4.16 (there seems to have been an issue where I set `VirtioFS` and it didn't truly take effect until a reboot. This problem still exists on a versions pointed to by https://github.com/docker/roadmap/issues/7#issuecomment-1357785737 when using `VirtioFS`.

**Comment by ericvn** [comment]
Also opened Case #: 00050830m with Docker support and https://github.com/docker/for-mac/issues/6651.

**Comment by ericvn** [comment]
Trying to build a new build-tool results in another error:
```
ERROR: mkdir /home/.docker: permission denied
make[1]: *** [Makefile.core.mk:41: containers-test] Error 1
make: *** [containers-test] Error 2
```

**Comment by fredericdalleau** [comment]
Here what I tested
```
fredo@letchi istio % make shell
<snip the warnings>
build-tools:/work$ mkdir /home/.docker
mkdir: cannot create directory '/home/.docker': Permission denied
build-tools:/work$ id -u
501
build-tools:/work$ ls -ld /home
drwxr-xr-x 1 root root 4096 Jan 12 17:33 /home
```
User 501 cannot write a directory that belongs to root.
Btw, this UID comes from the Mac host and is unknown to the VM.

**Comment by ericvn** [comment]
@fredericdalleau I did notice the 501 earlier today. I wasn't one of the original team that did the work of creating the `building-tools` image so I don't know all the things that they do within the code. I am learning as I see things with new Docker Desktop versions as we run into errors occasionally with new versions. For example, I did create a PR which would remove the warnings on 4.15 (although we never merged it (I was waiting to see if 4.16 actually removed the need for it)).

4.16.0 didn't fix it, but changed the behavior yet again. Now I'm trying to figure out how things changed (in gRPC and VirtioFS) and if there is something to fix our tooling.

Some pointers:
Dockerfile for the image: https://github.com/istio/tools/blob/master/docker/build-tools/Dockerfile
Entry-point code: https://github.com/istio/tools/blob/master/docker/build-tools/docker-entrypoint.sh

Doing some reading it seems like su-exec may not be something that should have been used.

**Comment by ericvn** [comment]
Appears to maybe be related to the use of `su-exec` and a behavior change in 4.16. In 4.15, when we run the container specifying a -u we can still run `su-exec 0:0 ls` as an example. This does not work in 4.16. If I remove the -u so the contianer runs as root, I can do the su-exec.

I do see on the su-exec page: `Notice that su-exec depends on being run by the root user, non-root users do not have permission to change uid/gid` so I'm not sure why things worked earlier.

**Comment by ericvn** [comment]
Pointer to Minikube having issues with 4.16: https://github.com/kubernetes/minikube/issues/15634

**Comment by ericvn** [comment]
Docker issue related to sudo: https://github.com/docker/for-mac/issues/6675

**Comment by ericvn** [comment]
The su-exec related issues seen in Docker For Mac 4.16.0 seem to be fixed in 4.16.1.
